docs/dictionary/command/filter.lcdoc

```diff
@@ -56,49 +56,51 @@ Parameters:
 filterSource:
 An expression that evaluates to a string, array, or a container.
 
+patternType:
+Either `wildcard pattern` or `regex pattern`.
```

Comment:
@montegoulding , is the wording here ok to you? Having it here in the parameters should draw more attention to it and it's explicit nature perhaps.

Reply:
I think that the original suggestion for the syntax is more accurate.  The patternType is only optional if `wildcard pattern` is the desired value.  Otherwise it is a required value.  A secondary consideration is that AutoComplete may no longer be able to suggest the pattern type with this change.

```
Syntax: filter [{lines | items | keys | elements} of] <filterSource> {with | without | [not] matching} {[wildcard pattern] | regex pattern} <filterPattern> [into <targetContainer>]
```

Reply:
Hmm 🤔. I’d like to defer up the chain on this one. It’s a good point and may well be the correct one. What are the thoughts. @runrevmark @livecodepanos @livecodeali @montegoulding anyone who wants to pipe in.

Reply:
I can definitely see the logic, although I'm not exactly convinced the syntax element should be used to imply something. Personally in terms of technical specification I find the existing syntax less confusing than the proposed new one, and would prefer the default behavior to be documented elsewhere (which I think you've done)

Reply:
Thanks @livecodeali. What about the difference between the original `[{wildcard pattern | regex pattern}]` and @bwmilby's suggestion of `{[wildcard pattern] | regex pattern}` which seems to have a more accurate logic to it?

Reply:
That's what I mean - to me the syntax element is just an expression of valid syntax in the language and shouldn't necessarily try to convey meaning in the way suggested - there's no actual difference between them in terms of what syntax is valid. 

I guess if people find `{[wildcard pattern] | regex pattern}` more useful than `[{wildcard pattern | regex pattern}]` then it's fair enough to make the change! As long as it doesn't have any negative consequences for autocomplete (I don't see why it would)

Reply:
On the other hand I guess we do split out syntactic variants into 'types' in some entries.
```
Syntax: filter [{lines | items | keys | elements} of] <filterSource> {with | without | [not] matching} [wildcard pattern] <filterPattern> [into <targetContainer>]
Syntax: filter [{lines | items | keys | elements} of] <filterSource> {with | without | [not] matching} regex pattern <filterPattern> [into <targetContainer>]
```
is also a possibility

Reply:
Ooo, now that is much better.

Reply:
@seaniepie I agree that splitting them is better. The filter command will only get more complicated when my patch for the `where` clause using the `each` global is merged https://github.com/livecode/livecode/pull/6626

Reply:
Actually splitting them gives us the option to use `<wildcardPattern>` and `<regularExpression>` on the different lines as the parameter names and document them separately.